DSCResources/MSFT_xVhdFileDirectory/MSFT_xVhdFileDirectory.psm1

```diff
@@ -365,74 +450,107 @@ function GetItemToCopy
 # This is the main function that gets called after the file is mounted to perform copy, set or new operations on the mounted drive.
 function SetVHDFile
 {
-     [CmdletBinding(DefaultParametersetName="Copy")] 
-    param(       
+    [CmdletBinding(DefaultParametersetName="Copy")] 
+    param
+    (
         [parameter(Mandatory=$true,ParameterSetName = "Copy")]
-        $sourcePath,        
+        $sourcePath,
+        
         [switch]$recurse,
+        
         [switch]$force,
+        
         [parameter(Mandatory=$false,ParameterSetName = "New")]  
         $type,
+        
         [parameter(Mandatory=$false,ParameterSetName = "New")]  
-        $content,       
+        $content,
+        
         [parameter(Mandatory=$true)]
-        $destinationPath, 
+        $destinationPath,
+        
         [parameter(Mandatory=$true,ParameterSetName = "Set")]  
         $attribute,
+        
         [parameter(Mandatory=$true,ParameterSetName = "Delete")]
-        [switch]$ensure 
-        )      
+        [switch]$ensure
+    )
     
-    Write-Verbose "Setting the VHD file $($PSCmdlet.ParameterSetName)"
-    if ($PSCmdlet.ParameterSetName -eq 'Copy')
+    Write-Verbose ($localizedData.VHDFileStatus -f $($PSCmdlet.ParameterSetName))
+    if($PSCmdlet.ParameterSetName -eq 'Copy')
     {
         New-Item -Path (Split-Path $destinationPath) -ItemType Directory -ErrorAction SilentlyContinue        
-        Copy-Item -Path $sourcePath -Destination $destinationPath -Force:$force -Recurse:$recurse -ErrorAction SilentlyContinue
+        try
+        {
+            Copy-Item -Path $sourcePath -Destination $destinationPath -Force:$force -Recurse:$recurse -ErrorAction SilentlyContinue
+            Write-Verbose ($localizedData.CopySuccessful -f $sourcePath,$destinationPath)
+        }
+        catch
+        {
+            Write-Verbose ($localizedData.CopyFailed -f $sourcePath,$destinationPath,$_.Exception.Message)
+        }
     }
-    elseif ($PSCmdlet.ParameterSetName -eq 'New')
+    elseif($PSCmdlet.ParameterSetName -eq 'New')
     {
-        If ($type -eq 'Directory')
+        if($type -eq 'Directory')
         {
             New-Item -Path $destinationPath -ItemType $type
         }
         else
         {
             New-Item -Path $destinationPath -ItemType $type
-            $content | Out-File $destinationPath 
+            $content | Out-File -FilePath $destinationPath 
         }
-
     }
-    elseif ($PSCmdlet.ParameterSetName -eq 'Set')
+    elseif($PSCmdlet.ParameterSetName -eq 'Set')
     {
-        Write-Verbose "Attempting to change the attribute of the file $destinationPath to value $attribute"
-        Set-ItemProperty -Path $destinationPath -Name Attributes -Value $attribute
+        try
+        {
+            Set-ItemProperty -Path $destinationPath -Name Attributes -Value $attribute -ErrorAction Stop
```

Comment:
These aren't error logs, just states that it worked if it goes past the Set-ItemProperty